preprocess/preprocess.go

```diff
@@ -153,30 +153,54 @@ func populateEnvMap(envMap map[string]map[string]interface{}, envPath string, so
 	return nil
 }
 
-func GetL2EnvVariables(dir string) ([]byte, error) {
+func getEnvMap(envPath string, source string) (map[string]map[string]interface{}, error) {
+	envs, err := readFile(envPath)
+	if err != nil {
+		return nil, err
+	}
+	envMap := make(map[string]map[string]interface{})
+	for key, value := range envs {
+		variable := make(map[string]interface{})
+		variable["val"] = value
+		variable["src"] = source
+		envMap[key] = variable
+	}
+	return envMap, nil
+}
+
+func combineEnvMaps(envMaps ...map[string]map[string]interface{}) map[string]map[string]interface{} {
```

Comment:
Add a comment regarding desired order for `l2config.env` and `l2.env` in this case. What overwrites what?